### ISSUE TITLE
Investigate internal JSON-RPC issues with metamask

### ISSUE THREAD

**Issue by aadam-10** [wormhole-foundation/wormhole-connect#827]
Reported in https://github.com/wormhole-foundation/wormhole-connect/issues/228

```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x9631288F4050F7CFbf77B77f8540DeCF6cfC7012","gasLimit":{"type":"BigNumber","hex":"0x03d090"},"to":"0x5a58505a96D1dbf8dF91cB21B54419FC36e93fdE","data":"0xc68785190000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000041801000000030d00ae06ccffbc521121320810d6e82f470fa79b15a1d30fafc6f529af29cbe168bf5bff2165ee214fe754f44d64aa9a7535bd3b6c6387bc0c4728a70aba42ebe1b7000171952cfafcb3ce21ea0ed748e3b944fc12f9cfcff4bf444d23667a72514e429218358e76cfc548feeb6cc054a0de50bb8ca971fe17a465147d2af833158a733a010241856e885f1b8278e6c7e55c22667bae725f979bf9521d2d1defbdf19170905c5965e461990426db9d9d81e9ebba2dd06706b0e24e946f699aad08bfe3ed1ea701039c0580fb2068b3754858c9895fc66ea1517b1f87a3ff5328acd76943eae5b12161f469efce500a5343cecd1dc332f2cc9d9b3591c465564c48c28e04ea6ab0350004fe0cceadf86e651727cb1557af92c3bbd719e98f8fb1556dd310e188ee24a12e5727339a9b05e33d767dfb86bb2dc216a7a1d402738181fcc589bf2b2eb04881000888d5fd62cc0f1fe746492d032fa06564f7a29ad78aeaef9271829ebd076ce7347e6872d6f29165968e7a18b57045948de039e3b45efb4ed5dfb64345d2343ea2000ab183649f684ce6d5c1a170403f6f7da7c1a78ea488fe782307bb31ce79a9062056208a3369d346f236e828f87c4a4a0e83059ec162440da143c8c491a6f6a889010c21c47d39a4d342ebde45e95fa9ecf897cf8ee9d24375f40b660796257f7acc5e769cd6e523b4e6792a23a50411bedeebef486089f9fb919011386dfdd013042e010d196ac284de71c5f8b9c2eceadfaf0ee9b456b5f08c3491946e23829ce2ff66df3bab41b1ffab1725f2cd4f6d9229e8cc7494761eb0cb6f8e6562b8fdbd868104000ed421fbea6198271b7bb0b9bd24ecc9658a3739d96fb287d95d64d150db09194326d9a2750aff72e7b3efdf2b083f142c021072ec98635e404ea7eb0afe75e79b0010973fdda770acbc4cc2b55b1f0c2e0437beacb7386b12eb9bb29cbc232939b401486eb3277750c922d83ba146d6b926b7771312de936419d77f3566da6962c236001163c29ea10f90f3dc3840ac4513ae79e1001fa4c92b7a86de894b36c27edeabfd70a09e531dd39c22978e2a283ee6e0ce69f849620384ccac0ca44062db78274b0112c3fc3e6f5bb06055d8e6129b3c33ba877a4f22ea105c1e38a05801a3ceca3b7409455e0db85465e73f62bc717c94427982a79d64bc94c0a63014e878dd8884840064afe0980000e8f10015ccceeb29348f71bdd22ffef43a2a19c1f5b5e17c5cca5411529120182672ade500000000000019460001000000000000000000000000000000000000000000000000000000000b80f2409258181f5ceac8dbffb7030890243caed69a9599d2886d957a9cb7656af3bdb300150000000000000000000000009631288f4050f7cfbf77b77f8540decf6cfc7012000500000000000000000000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":-32000,"message":"execution reverted"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": -32000,\n    \"message\": \"execution reverted\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n    at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:31538)\\n    at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:34478)\\n    at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:35088)\\n    at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:72700)\\n    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:73699\\n    at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-3.js:12:100465\"\n}\n  at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:31538)\n  at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:34478)\n  at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:35088)\n  at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:72700)\n  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:73699\n  at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-3.js:12:100465"}, code=CALL_EXCEPTION, version=providers/5.7.2)
```
TLDR: `error={"code":-32603,"message":"Internal JSON-RPC error."` ...

**Comment by tsadovska** [comment]
Issue is reproducible on https://preview.portal-bridge-ui.pages.dev/96d72fd6f0/cosmos/ [Mainnet] when sending 0.001 USDT from Osmosis to Base for the first time. Claiming fails with an error:
```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x9631288F4050F7CFbf77B77f8540DeCF6cfC7012","to":"0x8d2de8d2f73F1F4cAB472AC9A881C9b123C79627","data":"0xc68785190000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000041801000000030d008a08b16be56bf32768f2961514171397a01877044a16c4353420f325c14f61733e9b30ee0bc173e6de5d3db8c49d93bcff7b84a6cf7e1961ff41755d740f5ea800013dbe164afe6b614aae8359436efbeb7f3fbfd1ceb5ef4e21f361250ba3bd38e565d6c8228dfc90960c6daf9aca5c902fa789ca4b2405d8b8c04596f86d5e60a80102f4164db747f38ec935457c97c70c56752badcd23e40169dc95a7d37be81c46ef102fc99eaf10a3cfc55101e3198109e9eb0ce880c93aa9c1212447c60c307c5b0103bfe4239296164e5e099247af4f93e68d4c5a6b604d262c9fd3b66c1171976bed719e4d1586316b80500dd312c469f695d5ed4ba3bf8a260eed6bb9ed5c9db91b0004da3ab0f421a9b9af261f51c96166c6176e20f2a83ae9dfacdce9adc03170165233819bfdf2e698f7c350308fa1a0a99f29beab834b8eaa52018902466fdf468b0106dbd2cfe1d45e2b0ac1b1f940a3b3c2a653dc3d5fe02eb9f7ac462c38fdb08c7e30f466c1460fc6596e9bc1954b0bc64faa475d27f6ab528e02041b472ea207a6000789169d9e3a3cf277891dee4a871e677b5c7a5bba9321863478f8491cf69a481b08f4b75b4b11827e9cb4544f823730b77c8de4ec9ecaad3555c0d7dfeb458df8010d721d79d9fa4d8a16fa9d63e14652b197794894b0af46cfad12eb0cadb7b421a6780ab404f1b24e2e3c926f74aafb6db8e18b7d9635db21b7d8a3d437e656e41b010ea179f941757c9e479b9b441382809d3cd7b9cb2d6f471739ae863d898a5814f41ba4665e212344e729684554d9466bab323ab1a24334d38181dfe296ddc085f9010fd9c2c63a55c3d306fe44e81fbc055af8fea77c88b33e1fed97cea5404bf2906b78af8becbc72bdfb5e6a2763b809c97ae86128b199f5bd94fbc6c8a4ec9e54bb001040aa62dc5c20c3e4e2d9b950263890c63a1694606deeeb646db72a5f8124b8153259be0b41b6c136adc0ab8e44a34f203905545f1df13a525b1a04d40196a8c400116cf2704ea5a670d8bc56233180e113d720004911be7b84249b2a820e282c20333a461ab954c0b0c3edc81e252da1f1f14fbb9c3456cbbe5c84a31a1f724154e300120b82d0f7c13a78bfd37b06fe8c9288ba609db4fd3f8350226c0405c6f91a31415d8ed02e1a9a53b8966817371b80a058591f0f449625c95dec59d84d7ea6ef5d0064f734440000171d0c20aeb534c45c3049d380b9d9b966f9895f53abd4301bfaff407fa09dea8ae7a924000000000000004e000100000000000000000000000000000000000000000000000000000000000003e8000000000000000000000000dac17f958d2ee523a2206206994597c13d831ec700020000000000000000000000009631288f4050f7cfbf77b77f8540decf6cfc7012001e00000000000000000000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":-32000,"message":"header not found"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": -32000,\n    \"message\": \"header not found\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n    at new n (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:41774)\\n    at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:44714)\\n    at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:45324)\\n    at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:44213)\\n    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:45212\\n    at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:19:37935\"\n}\n  at new n (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:41774)\n  at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:44714)\n  at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:45324)\n  at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:44213)\n  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:45212\n  at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:19:37935"}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at f.makeError (main.js:37:4721120)
    at f.throwError (main.js:37:4721240)
    at y (main.js:37:4778136)
    at u.<anonymous> (main.js:37:4788320)
    at Generator.throw (<anonymous>)
    at s (main.js:37:4777467)
```
User can successfully claim asset after clicking on Claim button again. 

https://github.com/wormhole-foundation/wormhole-connect/assets/10746654/614bcbec-4072-4e3a-a5ca-637fe339b9a6

**Comment by tsadovska** [comment]
Issue is reproducible on https://cctp-testnet--wormhole-connect.netlify.app/ [Testnet] when sending 0.001 ETH (Arbitrum Goerli) from Arbitrum Goerli to BSC. Error is displayed in Console when user click on "Approve and proceed with transaction" button:
```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x0a370E7c5F7F944d451B48bc151a7BEC454bc185","to":"0x23908A62110e21C04F3A4e011d24F901F911744A","value":{"type":"BigNumber","hex":"0x038d7ea4c68000"},"data":"0x9981509f00000000000000000000000000000000000000000000000000000000000000040000000000000000000000000a370e7c5f7f944d451b48bc151a7bec454bc185000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b6d10000","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":-32000,"message":"execution reverted"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": -32000,\n    \"message\": \"execution reverted\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n    at new n (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:42930)\\n    at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:48227)\\n    at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:48837)\\n    at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:44213)\\n    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:45212\\n    at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:19:37935\"\n}\n  at new n (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:42930)\n  at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:48227)\n  at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-2.js:16:48837)\n  at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:44213)\n  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:45212\n  at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-3.js:19:37935"}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at f.makeError (main.14eb17a0.js:37:4721120)
    at f.throwError (main.14eb17a0.js:37:4721240)
    at y (main.14eb17a0.js:37:4778136)
    at u.<anonymous> (main.14eb17a0.js:37:4788320)
    at Generator.throw (<anonymous>)
    at s (main.14eb17a0.js:37:4777467)
```



Issue is reproducible when clicking on "Approve and proceed with transaction button" again. 


![image](https://github.com/wormhole-foundation/wormhole-connect/assets/10746654/6718eb86-b1c9-4ba5-941e-a327f4dbf174)

**Comment by tsadovska** [comment]
Tested OK on https://latest--wormhole-connect.netlify.app/ [Mainnet]. User can successfully claim asset on the first attempt. 
* Send from tx: [Mintscan](https://www.mintscan.io/osmosis/transactions/B22EA95A035361BBDB8E63C8EC2D5670D5E35575B7EC71901094E0F75165B17C)
* Send to tx: [Basescan](https://basescan.org/tx/0x17624e23fa54b8dd804058677da89ed33578031beec42da7b6fe5d5516fbca2c)

**Comment by tsadovska** [comment]
Tested OK on https://cctp-only--wormhole-connect.netlify.app/ [Mainnet]. The following transactions were successfully claimed from the first attempt: 

![Screenshot 2023-09-14 at 23 39 54](https://github.com/wormhole-foundation/wormhole-connect/assets/10746654/be623cd5-0e42-4edd-ae1f-28bdea05c2de)

**Comment by tsadovska** [comment]
Issue is no longer reproducible on https://wormhole-connect.netlify.app/ [Testnet].
Remaining issue: Transactions from Osmosis are not starting #778. 
Ticket is closed.
